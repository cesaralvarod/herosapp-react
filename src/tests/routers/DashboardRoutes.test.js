import React from "react";
import DashboardRoutes from "../../routers/DashboardRoutes";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../auth/AuthContext";

describe("Testing in DashboardRoutes", () => {
  test("Component should show correctly", () => {
    const contextValue = {
      dispatch: jest.fn(),
      user: {
        name: "Cesar",
        logged: "true",
      },
    };

    const wrapper = mount(
      <MemoryRouter>
        <AuthContext.Provider value={contextValue}>
          <DashboardRoutes />
        </AuthContext.Provider>
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find("nav.navbar span.nav-link").text().trim()).toBe(
      contextValue.user.name
    );
  });
});
