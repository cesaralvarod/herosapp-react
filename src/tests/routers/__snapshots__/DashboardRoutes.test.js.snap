// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in DashboardRoutes Component should show correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <button
      aria-controls="navbarToggler"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarToggler"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Heros App
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarToggler"
    >
      <ul
        className="navbar-nav mr-auto mt-2 mt-lg-0"
      >
        <li
          className="nav-item active"
        >
          <a
            aria-current="page"
            className="nav-item nav-link active"
            href="/marvel"
            onClick={[Function]}
            style={Object {}}
          >
            Marvel
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/dc"
            onClick={[Function]}
          >
            DC
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/search"
            onClick={[Function]}
          >
            Search
          </a>
        </li>
      </ul>
      <ul
        className="navbar-nav mt-2"
        style={
          Object {
            "marginLeft": "auto",
          }
        }
      >
        <li
          className="nav-item"
        >
          <span
            className="text-info nav-item nav-link"
          >
            Cesar
          </span>
        </li>
        <li
          className="nav-item"
        >
          <button
            className="btn nav-item nav-link"
            onClick={[Function]}
          >
            Logout
          </button>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="container mt-2"
  >
    <h1>
      Marvel heros
    </h1>
    <hr />
    <div
      className="row row-cols-1 row-cols-md-2 g-4 mb-5 animate__animated animate__bounceInLeft"
    >
      <div
        className="col"
        key="marvel-spider"
      >
        <div
          className="card ms-3 mt-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters "
          >
            <div
              className="col-md-4 "
            >
              <img
                alt="Spider Man"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Spider Man
                </h5>
                <p
                  className="card-text"
                >
                  Peter Parker
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Amazing Fantasy #15
                  </small>
                </p>
                <a
                  href="/hero/marvel-spider"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col"
        key="marvel-captain"
      >
        <div
          className="card ms-3 mt-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters "
          >
            <div
              className="col-md-4 "
            >
              <img
                alt="Captain America"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Captain America
                </h5>
                <p
                  className="card-text"
                >
                  Steve Rogers
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Captain America Comics #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-captain"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col"
        key="marvel-iron"
      >
        <div
          className="card ms-3 mt-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters "
          >
            <div
              className="col-md-4 "
            >
              <img
                alt="Iron Man"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Iron Man
                </h5>
                <p
                  className="card-text"
                >
                  Tony Stark
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Tales of Suspense #39
                  </small>
                </p>
                <a
                  href="/hero/marvel-iron"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col"
        key="marvel-thor"
      >
        <div
          className="card ms-3 mt-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters "
          >
            <div
              className="col-md-4 "
            >
              <img
                alt="Thor"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Thor
                </h5>
                <p
                  className="card-text"
                >
                  Thor Odinson
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Journey into Myster #83
                  </small>
                </p>
                <a
                  href="/hero/marvel-thor"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col"
        key="marvel-hulk"
      >
        <div
          className="card ms-3 mt-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters "
          >
            <div
              className="col-md-4 "
            >
              <img
                alt="Hulk"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Hulk
                </h5>
                <p
                  className="card-text"
                >
                  Bruce Banner
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    The Incredible Hulk #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-hulk"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col"
        key="marvel-wolverine"
      >
        <div
          className="card ms-3 mt-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters "
          >
            <div
              className="col-md-4 "
            >
              <img
                alt="Wolverine"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Wolverine
                </h5>
                <p
                  className="card-text"
                >
                  James Howlett
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    The Incredible Hulk #180
                  </small>
                </p>
                <a
                  href="/hero/marvel-wolverine"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col"
        key="marvel-daredevil"
      >
        <div
          className="card ms-3 mt-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters "
          >
            <div
              className="col-md-4 "
            >
              <img
                alt="Daredevil"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Daredevil
                </h5>
                <p
                  className="card-text"
                >
                  Matthew Michael Murdock
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  />
                </p>
                <a
                  href="/hero/marvel-daredevil"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col"
        key="marvel-hawkeye"
      >
        <div
          className="card ms-3 mt-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters "
          >
            <div
              className="col-md-4 "
            >
              <img
                alt="Hawkeye"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Hawkeye
                </h5>
                <p
                  className="card-text"
                >
                  Clinton Francis Barton
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Tales of Suspense #57
                  </small>
                </p>
                <a
                  href="/hero/marvel-hawkeye"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col"
        key="marvel-cyclops"
      >
        <div
          className="card ms-3 mt-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters "
          >
            <div
              className="col-md-4 "
            >
              <img
                alt="Cyclops"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Cyclops
                </h5>
                <p
                  className="card-text"
                >
                  Scott Summers
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    X-Men #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-cyclops"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col"
        key="marvel-silver"
      >
        <div
          className="card ms-3 mt-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters "
          >
            <div
              className="col-md-4 "
            >
              <img
                alt="Silver Surfer"
                className="card-img"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Silver Surfer
                </h5>
                <p
                  className="card-text"
                >
                  Norrin Radd
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    The Fantastic Four #48
                  </small>
                </p>
                <a
                  href="/hero/marvel-silver"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
